src/DynamoAnalyzer/Helper/CsvHandler.cs

```diff
@@ -0,0 +1,151 @@
+                await csv.WriteRecordsAsync(packages);
+            }
+
+            return file.FullName;
+        }
+    }
+
+    /// <summary>
+    /// Defines the properties to be written to a file
+    /// </summary>
+    public class DuplicatedPackageMap : ClassMap<DuplicatedPackage>
+    {
+        public DuplicatedPackageMap()
+        {
+            Map(m => m.ArtifactName).Name(nameof(DuplicatedPackage.ArtifactName));
+            Map(m => m.Count).Name(nameof(DuplicatedPackage.Count));
+            Map(m => m.Packages).TypeConverter<ArrayConverter>();
+        }
+    }
+
+    /// <summary>
+    /// Defines the properties to be written to a file
+    /// </summary>
+    public class AnalyzedPackageMap : ClassMap<AnalyzedPackage>
+    {
+        public AnalyzedPackageMap()
+        {
+            Map(m => m.Index).Name(nameof(AnalyzedPackage.Index));
+            Map(m => m.Id).Name(nameof(AnalyzedPackage.Id));
+            Map(m => m.Name).Name(nameof(AnalyzedPackage.Name));
+            Map(m => m.Version).Name(nameof(AnalyzedPackage.Version));
+
+            Map(m => m.UserId).Name(nameof(AnalyzedPackage.UserId));
+            Map(m => m.UserName).Name(nameof(AnalyzedPackage.UserName));
+
+            Map(m => m.ArchiveName).Name(nameof(AnalyzedPackage.ArchiveName));
+            Map(m => m.ArtifactPath).Name(nameof(AnalyzedPackage.ArtifactPath));
+            Map(m => m.ArtifactName).Name(nameof(AnalyzedPackage.ArtifactName));
+            Map(m => m.HasBinaries).Name(nameof(AnalyzedPackage.HasBinaries));
+            Map(m => m.HasSource).Name(nameof(AnalyzedPackage.HasSource));
+            Map(m => m.RequirePort).Name(nameof(AnalyzedPackage.RequirePort));
+            Map(m => m.HasAnalysisError).Name(nameof(AnalyzedPackage.HasAnalysisError));
+            Map(m => m.DLLs).Ignore();
+            Map(m => m.Source).Ignore();
+            Map(m => m.Result).TypeConverter<ResultArrayConverter>();
+        }
+    }
+
+    /// <summary>
+    /// Converts an string array to it's string representation
+    /// </summary>
+    public class ArrayConverter : DefaultTypeConverter
+    {
+        public override string ConvertToString(object value, IWriterRow row, MemberMapData memberMapData)
+        {
+            if (value.GetType().Name.Equals("String[]"))
+            {
+                return JsonConvert.SerializeObject(value).Replace(",", "&").Replace("\"", "'");
+            }
+            return "";
+        }
+    }
+
+    /// <summary>
+    /// Converts an string array to it's string representation
+    /// </summary>
+    public class ResultArrayConverter : DefaultTypeConverter
+    {
+        public override string ConvertToString(object value, IWriterRow row, MemberMapData memberMapData)
+        {
+            if (value.GetType().Name.Equals("String[]"))
+            {
+                string[] values = (string[])value;
+                string[] valuesToWrite = new string[6];
```

Comment:
?? Why are there 6 strings, seems fragile etc.

Reply:
For each dll it could be up to 50 or more references suggestions from the upgrade-assistant tool stored in the results property, which would generate a huge text for a cell when converting to csv, which also could break the columns when opening the csv file in Microsoft Excel, so I limited the suggestions to 5 and set the property RequirePort to true